### PR TITLE
Add lint checks for exec bit and json validity

### PR DESCRIPTION
**Pull request by ssbarnea**
- ensure executable bits are not misused
- check validity of pure json files

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
 	  "moduleResolution": "node",
 	  "outDir": "./out/server",
 	  "sourceMap": true,
-	  "target": "es6",
```
Comment (webknjaz) [inline tsconfig.json:11]: No need to drop trailing commas from tsconfig — Typescript is smart enough to allow them, and they make diffs nicer.

**Review by webknjaz — CHANGES_REQUESTED** [review]
Plz don't use commit prefixes. They make the log ugly. Besides, without it, you could come up with a useful commit message having more free characters.

File: tsconfig.json (hunk 1 above)
Comment (webknjaz) [reply tsconfig.json:11]: (I suppose you've excluded this file in the pre-commit config for exactly this reason)

Comment (ssbarnea) [reply tsconfig.json:11]: In fact that file is not excluded, another one with same name but different path is. This change was required by `check-json`.

That story around json-with-comments makes the json-check almost useless. Do you want me to exclude it and avoid the change? I guess that if it was broken we would have known anyway.

Comment (ssbarnea) [reply tsconfig.json]: done

**Comment by ssbarnea** [comment]
> Plz don't use commit prefixes. They make the log ugly. Besides, without it, you could come up with a useful commit message having more free characters.

these prefixes are already used in vscode-ansible and even enforced. For the cost of 3-4 chars it provides more semantic
commits with some extra benefits: automatic labeling or PR (which non cores cannot do) and also ability to make commits easier to read and even sort by category.

Explaining if a change is major, minor, patch or to be ignored cannot be achieved without prefixes.

You should remark that when prefix is present you no longer need to repeat yourself. In some cases the summary may be even shorter than without the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

Example:
* `Fix comments in .foo files`
* `fix: comments in .foo files`

Probably we should discuss this after we return from vacation. If PR name is what prevents this to merge just rename it and approve it, is more important to get those fixed in than agreeing on ideal naming for changes.

File: tsconfig.json (hunk 1 above)
Comment (webknjaz) [reply tsconfig.json]: Yep, we'll see tsc failures if it stops understanding it.

**Comment by webknjaz** [comment]
Yes, we will need to discuss removing conventional-commits from there as well.

**Comment by webknjaz** [comment]
But please fix the PR title yourself. Just removing the prefix is not enough. It would still be pointless for the history in this state.

**Review by webknjaz — APPROVED** [review]

**Pull request merged**